homeassistant/components/fibaro/binary_sensor.py

```diff
@@ -38,54 +41,45 @@ def setup_platform(
     if discovery_info is None:
         return
 
-    add_entities(
-        [
-            FibaroBinarySensor(device)
-            for device in hass.data[FIBARO_DEVICES]["binary_sensor"]
-        ],
-        True,
-    )
+    entities: list[FibaroBinarySensor] = []
+    for fibaro_device in hass.data[FIBARO_DEVICES]["binary_sensor"]:
+        device_class = None
+        icon = None
+        if fibaro_device.type in SENSOR_TYPES:
+            device_class = SENSOR_TYPES[fibaro_device.type][2]
+            icon = SENSOR_TYPES[fibaro_device.type][1]
+        elif fibaro_device.baseType in SENSOR_TYPES:
+            device_class = SENSOR_TYPES[fibaro_device.baseType][2]
+            icon = SENSOR_TYPES[fibaro_device.baseType][1]
+
+        # device_config overrides:
+        devconf = fibaro_device.device_config
+        device_class = devconf.get(CONF_DEVICE_CLASS, device_class)
+        icon = devconf.get(CONF_ICON, icon)
+
+        entity_description = BinarySensorEntityDescription(
+            key="binary_sensor",
+            name=fibaro_device.friendly_name,
+            device_class=device_class,
+            icon=icon,
+        )
```

Comment:
There is not much use in using entity descriptions like this. I would suggest keeping it in the binary sensor class or actually defining all binary sensors upfront.

Reply:
Thanks for your comment. I agree that this probably does not make much sense. So looks like using EntityDescription is not mandatory and therefore we should keep the logic here as is. Therefore I will decline this PR as it looks like it is not needed.